piqe_ocp_lib/api/resources/ocp_nodes.py

```diff
@@ -80,18 +80,47 @@ def get_total_memory_in_bytes(self):
             logger.info("Total memory in bytes : %s", total_memory_in_bytes)
         return total_memory_in_bytes
 
-    def watch_all_nodes(self):
-        # We need to determine if a use case for this exists.
-        pass
-
-    def watch_a_node(self, node_name):
-        pass
-
-    def create_a_node(self, node_name):
-        pass
-
-    def delete_a_node(self, node_name):
-        pass
+    def is_node_ready(self, node_name, timeout=300):
+        """
+        Check if a node has reached a Ready state
+        :param node_name: (str) The node name
+        :param timeout: (int) The time limit for polling status. Defaults to 300
+        :return: (bool) True if it's Ready OR False otherwise
+        """
+        field_selector = "metadata.name={}".format(node_name)
+        for event in self.ocp_nodes.watch(field_selector=field_selector, timeout=timeout):
+            conditions_list = event['object']['status']['conditions']
+            latest_event = conditions_list[-1]
+            if conditions_list and latest_event['type'] == "Ready" and latest_event['status'] == "True":
+                logger.debug("Node {} has reached 'Ready' state".format(node_name))
+                return True
+            else:
+                logger.debug("Waiting for node {} to reach 'Ready' state."
+                             "Reason: {}".format(node_name, latest_event['message']))
+        return False
+
+    def is_node_deleted(self, node_name, timeout=300):
```

Comment:
We'll need to introduce tests for this. We can capture this in a separate PR.

Reply:
Yes, see my response about testing watch type methods.